latch/ldata/path.py

```diff
@@ -0,0 +1,171 @@
+from pathlib import Path
+from typing import Generator, Optional, Union
+from urllib.parse import urljoin
+from uuid import uuid4
+
+import gql
+from latch_sdk_config.latch import NUCLEUS_URL
+from latch_sdk_gql.execute import execute
+
+from latch.ldata.node import (
+    LDataNodeType,
+    LDataPerms,
+    PermLevel,
+    get_node_data,
+    get_node_metadata,
+    get_node_perms,
+)
+from latch.ldata.transfer import download, remote_copy, upload
+from latch.ldata.transfer.progress import Progress
+from latch_cli.tinyrequests import post
+from latch_cli.utils import get_auth_header, urljoins
+from latch_cli.utils.path import is_remote_path
+
+
+class LPath:
+    def __init__(self, path: str):
+        if not is_remote_path(path):
+            raise ValueError(f"Invalid LPath: {path} is not a Latch path")
+        self.path = path
+        self._node_id = None
+
+    @property
+    def node_id(self) -> str:
```

Comment:
don't use `property` on things that might make network requests

use a parameter to allow/forbid network requests if data is its required (see Registry APIs for examples)

---

try to parse the path as a `^latch://(?P<id>[0-9]+).node$`